### PR TITLE
Fix rehash bug in TypedHistogram and added test

### PR DESCRIPTION
**Pull request by raghavsethi**

### CODE REVIEW

**Comment by haozhun** [comment]
checkstyle failure

File: presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+
+public class TestTypedHistogram
+{
+    @Test
+    public void testMassive()
+        throws Exception
+    {
+        TypedHistogram typedHistogram = new TypedHistogram(BIGINT, 1000);
+        List<Object> input = IntStream.range(1, 2000).mapToObj(i -> IntStream.iterate(i, j -> j).limit(i).mapToObj(k -> k)).flatMap(l -> l).collect(Collectors.toList());
```
Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java]: format like this
```
List<Object> input = IntStream.range(1, 2000)
        .mapToObj(i -> IntStream.iterate(i, j -> j).limit(i).mapToObj(k -> k))
        .flatMap(l -> l)
        .collect(Collectors.toList());
```

Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java]: Use `.boxed` instead of `.mapToObj(k -> k)`

Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java]: Use `Function.identity()`

Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java]: Use `flatMapToInt`

Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java]: Additionally, you don't need to collect them into a list and then loop over the list. You can do `forEach` on the stream.

**Comment by raghavsethi** [comment]
Weird, I can't see the checkstyle issue..

**Comment by raghavsethi** [comment]
@haozhun: Please re-review when you get a chance, we need to get this out in this release for sure.

File: presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        Block inputBlock = inputBlockBuilder.build();
+        for (int i = 0; i < inputBlock.getPositionCount(); i++) {
+            typedHistogram.add(i, inputBlock, 1);
+        }
+
+        Block outputBlock = typedHistogram.serialize();
+        for (int i = 0; i < outputBlock.getPositionCount(); i += 2) {
+            if (!BIGINT.equalTo(outputBlock, i, outputBlock, i + 1)) {
+                Assert.assertEquals(BIGINT.getLong(outputBlock, i + 1), BIGINT.getLong(outputBlock, i));
```
Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java]: static import

File: presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                .forEach(j ->  BIGINT.writeLong(inputBlockBuilder, j));
+
+        Block inputBlock = inputBlockBuilder.build();
+        for (int i = 0; i < inputBlock.getPositionCount(); i++) {
+            typedHistogram.add(i, inputBlock, 1);
+        }
+
+        Block outputBlock = typedHistogram.serialize();
+        for (int i = 0; i < outputBlock.getPositionCount(); i += 2) {
+            if (!BIGINT.equalTo(outputBlock, i, outputBlock, i + 1)) {
```
Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java]: remove this if?

File: presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+{
+    @Test
+    public void testMassive()
+        throws Exception
+    {
+        BlockBuilder inputBlockBuilder = BIGINT.createBlockBuilder(new BlockBuilderStatus(), 5000);
+
+        TypedHistogram typedHistogram = new TypedHistogram(BIGINT, 1000);
+        IntStream.range(1, 2000)
+                .mapToObj(i -> IntStream.iterate(i, IntUnaryOperator.identity()).limit(i).boxed())
```
Comment (erichwang) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java]: I think you can use IntStream.rangeClosed(i, i+i)

File: presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                .flatMap(Function.identity())
```
Comment (erichwang) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestTypedHistogram.java]: you can merge the above mapToObj to this flatMap

**Comment by erichwang** [comment]
looks good otherwise

**Pull request merged**